clusterloader2/pkg/measurement/common/exec.go

```diff
@@ -62,22 +62,40 @@ func (e *execMeasurement) Execute(config *measurement.Config) ([]measurement.Sum
 	if len(command) == 0 {
 		return nil, fmt.Errorf("command is a required argument. Got empty slice instead")
 	}
-
+	retries, err := util.GetIntOrDefault(config.Params, "retries", 1)
+	if err != nil || retries < 1 {
+		return nil, fmt.Errorf("error getting retries, retries: %v, err: %v", retries, err)
+	}
+	retryIntervalStr, err := util.GetStringOrDefault(config.Params, "retryInterval", "1s")
+	if err != nil {
+		return nil, err
+	}
+	retryInterval, err := time.ParseDuration(retryIntervalStr)
+	if err != nil {
+		return nil, err
+	}
 	// Make a copy of command, to avoid overriding a slice we don't own.
 	command = append([]string{}, command...)
 	for i := range command {
 		command[i] = os.ExpandEnv(command[i])
 	}
-	klog.V(2).Infof("Running %v with timeout %v", command, timeout)
-	ctx, cancel := context.WithTimeout(context.Background(), timeout)
-	defer cancel()
-	cmd := exec.CommandContext(ctx, command[0], command[1:]...)
-	out, err := cmd.CombinedOutput()
-	klog.V(2).Infof("Exec command output: %v", string(out))
-	if err != nil {
-		return nil, fmt.Errorf("command %v failed: %v", command, err)
+	var lastErr error
+	for i := 0; i < retries; i++ {
+		klog.V(2).Infof("Running %v with timeout %v, attempt %v", command, timeout, i+1)
+		ctx, cancel := context.WithTimeout(context.Background(), timeout)
+		defer cancel()
+		cmd := exec.CommandContext(ctx, command[0], command[1:]...)
+		out, err := cmd.CombinedOutput()
+		klog.V(2).Infof("Exec command output: %v", string(out))
+		if err == nil {
+			klog.V(2).Infof("Command %v succeeded in attempt %v", command, i+1)
+			return nil, nil
+		}
```

Comment:
I'd probably still print out the errors if the past retries failed.

Reply:
Done